usaspending_api/conftest_helpers.py

```diff
@@ -36,45 +42,65 @@ def update_index(self):
         for the index, and add contents.
         """
         self.delete_index()
-        self._refresh_materialized_views()
+        self._refresh_materialized_views(self.type)
         self.client.indices.create(self.index_name, self.template)
-        create_aliases(self.client, self.index_name, True)
+        create_aliases(self.client, self.index_name, True, awards=(self.type == "awards"))
         self._add_contents()
 
     def _add_contents(self):
         """
         Get all of the transactions presented in the view and stuff them into the Elasticsearch index.
         The view is only needed to load the transactions into Elasticsearch so it is dropped after each use.
         """
-        transaction_delta_view_sql = open(
-            str(settings.APP_DIR / "database_scripts" / "etl" / "transaction_delta_view.sql"), "r"
-        ).read()
+        view_sql = (
+            open(str(settings.APP_DIR / "database_scripts" / "etl" / "award_delta_view.sql"), "r").read()
+            if self.type == "awards"
+            else open(str(settings.APP_DIR / "database_scripts" / "etl" / "transaction_delta_view.sql"), "r").read()
+        )
         with connection.cursor() as cursor:
-            cursor.execute(transaction_delta_view_sql)
-            cursor.execute(f"SELECT * FROM {settings.ES_TRANSACTIONS_ETL_VIEW_NAME};")
-            transactions = ordered_dictionary_fetcher(cursor)
-            cursor.execute(f"DROP VIEW {settings.ES_TRANSACTIONS_ETL_VIEW_NAME};")
-
-        for transaction in transactions:
-            self.client.index(
-                self.index_name,
-                self.doc_type,
-                json.dumps(transaction, cls=DjangoJSONEncoder),
-                transaction["transaction_id"],
-            )
+            cursor.execute(view_sql)
+            if self.type == "transactions":
```

Comment:
Most of the nested code is shared. It would be cleaner to set variables for the differences in view name and ID field and then execute the commands and indexing.